### PR TITLE
Add suppoort for Bypass By Default mode for the Edge TTL block in Rulesets

### PR DESCRIPTION
**Pull request by iveelsm**
Pretty much the exact same as https://github.com/cloudflare/terraform-provider-cloudflare/pull/2756

Just with a different mode for Edge TTL

### CODE REVIEW

**Comment by jacobbednarz** [comment]
looks like this one is failing on one of the expected failure scenarios?

```
TF_ACC=1 go test ./internal/framework/service/rulesets -v -run "^TestAccCloudflareRuleset_Cache" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareRuleset_CacheSettingsAllEnabled
--- PASS: TestAccCloudflareRuleset_CacheSettingsAllEnabled (11.54s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsOptionalsEmpty
--- PASS: TestAccCloudflareRuleset_CacheSettingsOptionalsEmpty (10.10s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsEdgeTTLRespectOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsEdgeTTLRespectOrigin (10.37s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsNoCacheForStatus
--- PASS: TestAccCloudflareRuleset_CacheSettingsNoCacheForStatus (10.73s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsStatusRangeGreaterThan
--- PASS: TestAccCloudflareRuleset_CacheSettingsStatusRangeGreaterThan (10.60s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsStatusRangeLessThan
--- PASS: TestAccCloudflareRuleset_CacheSettingsStatusRangeLessThan (10.56s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsFalse
--- PASS: TestAccCloudflareRuleset_CacheSettingsFalse (10.15s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsMissingEdgeTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsMissingEdgeTTLWithOverrideOrigin (6.11s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsMissingBrowserTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsMissingBrowserTTLWithOverrideOrigin (6.17s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithOverrideOrigin (6.12s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsEdgeTTLWithBypassByDefault
    resource_test.go:2319: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: invalid configuration
        
          with cloudflare_ruleset.pnlkshawzy,
          on terraform_plugin_test.tf line 15, in resource "cloudflare_ruleset" "pnlkshawzy":
          15: 			edge_ttl {
        
        cannot set default ttl when using mode 'bypass_by_default'
--- FAIL: TestAccCloudflareRuleset_CacheSettingsEdgeTTLWithBypassByDefault (6.14s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithBypassByDefault
    resource_test.go:2346: Step 1/1, expected an error with pattern, no match on: Error running apply: exit status 1
        
        Error: error creating ruleset csxredarjx
        
          with cloudflare_ruleset.csxredarjx,
          on terraform_plugin_test.tf line 2, in resource "cloudflare_ruleset" "csxredarjx":
           2: 	resource "cloudflare_ruleset" "csxredarjx" {
        
        default is useless in bypass_by_default mode (20107)
--- FAIL: TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithBypassByDefault (7.87s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsBrowserTTLWithBypass
--- PASS: TestAccCloudflareRuleset_CacheSettingsBrowserTTLWithBypass (10.40s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithBypass
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithBypass (6.07s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithOverrideOrigin (6.04s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringExcludeKeys
--- PASS: TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringExcludeKeys (10.47s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys
--- PASS: TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys (10.61s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsHandleDefaultHeaderExcludeOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsHandleDefaultHeaderExcludeOrigin (15.77s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/rulesets	166.506s
FAIL
make: *** [testacc] Error 1
```

**Comment by iveelsm** [comment]
> looks like this one is failing on one of the expected failure scenarios?
> 
> ```
> TF_ACC=1 go test ./internal/framework/service/rulesets -v -run "^TestAccCloudflareRuleset_Cache" -count 1 -timeout 120m -parallel 1
> === RUN   TestAccCloudflareRuleset_CacheSettingsAllEnabled
> --- PASS: TestAccCloudflareRuleset_CacheSettingsAllEnabled (11.54s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsOptionalsEmpty
> --- PASS: TestAccCloudflareRuleset_CacheSettingsOptionalsEmpty (10.10s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsEdgeTTLRespectOrigin
> --- PASS: TestAccCloudflareRuleset_CacheSettingsEdgeTTLRespectOrigin (10.37s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsNoCacheForStatus
> --- PASS: TestAccCloudflareRuleset_CacheSettingsNoCacheForStatus (10.73s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsStatusRangeGreaterThan
> --- PASS: TestAccCloudflareRuleset_CacheSettingsStatusRangeGreaterThan (10.60s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsStatusRangeLessThan
> --- PASS: TestAccCloudflareRuleset_CacheSettingsStatusRangeLessThan (10.56s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsFalse
> --- PASS: TestAccCloudflareRuleset_CacheSettingsFalse (10.15s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsMissingEdgeTTLWithOverrideOrigin
> --- PASS: TestAccCloudflareRuleset_CacheSettingsMissingEdgeTTLWithOverrideOrigin (6.11s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsMissingBrowserTTLWithOverrideOrigin
> --- PASS: TestAccCloudflareRuleset_CacheSettingsMissingBrowserTTLWithOverrideOrigin (6.17s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithOverrideOrigin
> --- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithOverrideOrigin (6.12s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsEdgeTTLWithBypassByDefault
>     resource_test.go:2319: Step 1/1 error: Error running pre-apply refresh: exit status 1
>         
>         Error: invalid configuration
>         
>           with cloudflare_ruleset.pnlkshawzy,
>           on terraform_plugin_test.tf line 15, in resource "cloudflare_ruleset" "pnlkshawzy":
>           15: 			edge_ttl {
>         
>         cannot set default ttl when using mode 'bypass_by_default'
> --- FAIL: TestAccCloudflareRuleset_CacheSettingsEdgeTTLWithBypassByDefault (6.14s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithBypassByDefault
>     resource_test.go:2346: Step 1/1, expected an error with pattern, no match on: Error running apply: exit status 1
>         
>         Error: error creating ruleset csxredarjx
>         
>           with cloudflare_ruleset.csxredarjx,
>           on terraform_plugin_test.tf line 2, in resource "cloudflare_ruleset" "csxredarjx":
>            2: 	resource "cloudflare_ruleset" "csxredarjx" {
>         
>         default is useless in bypass_by_default mode (20107)
> --- FAIL: TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithBypassByDefault (7.87s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsBrowserTTLWithBypass
> --- PASS: TestAccCloudflareRuleset_CacheSettingsBrowserTTLWithBypass (10.40s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithBypass
> --- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithBypass (6.07s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithOverrideOrigin
> --- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithOverrideOrigin (6.04s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringExcludeKeys
> --- PASS: TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringExcludeKeys (10.47s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys
> --- PASS: TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys (10.61s)
> === RUN   TestAccCloudflareRuleset_CacheSettingsHandleDefaultHeaderExcludeOrigin
> --- PASS: TestAccCloudflareRuleset_CacheSettingsHandleDefaultHeaderExcludeOrigin (15.77s)
> FAIL
> FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/rulesets	166.506s
> FAIL
> make: *** [testacc] Error 1
> ```

There was a bug in the implementation. This has been addressed.

File: .changelog/2764.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/cloudflare_ruleset: Add support for bypass by default in Edge TTL modes
```
Comment (jacobbednarz) [inline .changelog/2764.txt:2]:
```suggestion
resource/cloudflare_ruleset: Add support for cache bypass by default in Edge TTL modes
```

**Comment by jacobbednarz** [comment]
acceptance tests all working now, thanks!

```
TF_ACC=1 go test ./internal/framework/service/rulesets -v -run "^TestAccCloudflareRuleset_Cache" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareRuleset_CacheSettingsAllEnabled
--- PASS: TestAccCloudflareRuleset_CacheSettingsAllEnabled (12.69s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsOptionalsEmpty
--- PASS: TestAccCloudflareRuleset_CacheSettingsOptionalsEmpty (11.27s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsEdgeTTLRespectOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsEdgeTTLRespectOrigin (10.91s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsNoCacheForStatus
--- PASS: TestAccCloudflareRuleset_CacheSettingsNoCacheForStatus (16.59s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsStatusRangeGreaterThan
--- PASS: TestAccCloudflareRuleset_CacheSettingsStatusRangeGreaterThan (10.42s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsStatusRangeLessThan
--- PASS: TestAccCloudflareRuleset_CacheSettingsStatusRangeLessThan (34.40s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsFalse
--- PASS: TestAccCloudflareRuleset_CacheSettingsFalse (11.84s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsMissingEdgeTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsMissingEdgeTTLWithOverrideOrigin (5.96s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsMissingBrowserTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsMissingBrowserTTLWithOverrideOrigin (21.70s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithOverrideOrigin (8.44s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsEdgeTTLWithBypassByDefault
--- PASS: TestAccCloudflareRuleset_CacheSettingsEdgeTTLWithBypassByDefault (25.90s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithBypassByDefault
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidEdgeTTLWithBypassByDefault (7.91s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsBrowserTTLWithBypass
--- PASS: TestAccCloudflareRuleset_CacheSettingsBrowserTTLWithBypass (11.38s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithBypass
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithBypass (6.33s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithOverrideOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsInvalidBrowserTTLWithOverrideOrigin (7.82s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringExcludeKeys
--- PASS: TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringExcludeKeys (12.03s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys
--- PASS: TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys (12.80s)
=== RUN   TestAccCloudflareRuleset_CacheSettingsHandleDefaultHeaderExcludeOrigin
--- PASS: TestAccCloudflareRuleset_CacheSettingsHandleDefaultHeaderExcludeOrigin (18.33s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/rulesets	247.714s
```

fixed the docs in 07a5b49 to match the expected output.

**Pull request merged**